config/routes.rb

```diff
@@ -443,8 +443,8 @@
         get '/type/edit/:id' => 'references/type#edit', as: :references_edit_type
         patch '/type/edit/:id' => 'references/type#update'
 
-        get '/name/:id' => 'references/name#new', as: :references_name
-        patch '/name/:id' => 'references/name#create'
+        get '/name/:referee_type' => 'references/name#new', as: :references_name
+        patch '/name/:referee_type' => 'references/name#create'
```

Comment:
Might be worth adding some constraints on the `:referee_type` here?

I managed to cause an exception by visiting `/candidate/application/references/name/invalid` and then filling out the form with a name and pressing save. Unlikely to happen in production though admittedly! Alternatively, could use a query string param and validate it?

```suggestion
        constraints: { referee_type: /(academic|professional|school-based|character)/ } do 
          get '/name/:referee_type' => 'references/name#new', as: :references_name
          patch '/name/:referee_type' => 'references/name#create'
        end
```

Reply:
A route constraint's a good idea, and probably sufficient  —  no need for graceful error handling and validation messages in this scenario IMHO

Reply:
@davidgisbey ☝️ I've implemented this above and it prevents the problem below. Will push up shortly.

Reply:
Yea this would work 👍